XMonad/Util/ExtensibleState.hs

```diff
@@ -121,8 +123,11 @@ remove :: (ExtensionClass a, XLike m) => a -> m ()
 remove wit = modifyStateExts $ M.delete (show . typeOf $ wit)
 
 modified :: (ExtensionClass a, Eq a, XLike m) => (a -> a) -> m Bool
-modified f = do
+modified = modifiedM . (pure .)
```

Comment:
Arguably this refactor should be a separate PR.

Reply:
I can definitely turn this into a commit on its own but a separate PR seems excessive.

Reply:
> Other than that (and a similar feeling about the `deleteFromStru[c]tCache` correction), this seems okay.

I split both into separate commits.